cEP-0016.md

```diff
@@ -0,0 +1,180 @@
+| Authors  | Kriti Rohillla <mailto:kritirohilla567@gmail.com> |
+| Status   | Implementation due                                |
+| Type     | Process                                           |
+
+## Abstract
+
+This document describes coala process rules which will be enforced by the
+GitCommitBear. Some new changes will be introduced in the implementation of
+GitCommitBear based on which the commits made to coala will be detected and
+verified.
+
+## Introduction
+
+The GitCommitBear at present performs a check on the content of regular commits
+made to coala. However, there are many special types of commit messages that
+should be used only in conjunction with patches containing a special type of
+content.
+
+For example, git revert commit which is used to revert any previous commit and
+git merge commit which is generated whenever someone performs a git rebase and
+tries to merge master onto the pull request. Such commits generate a default
+commit message that is very concise and not according to coala’s standards.
+The coala project wants to allow `git revert` commits, but does not want to
+accept `git merge` commits as the repositories are strictly fast-forward merges
+that do not have an extra commit.
+
+Also in cases where some tests are not required by any pull request made to
+coala, some special sequences can be added to git commit messages to instruct
+tools to operate in a certain way.
+
+This project is about detecting such commits and verifying that they are correct
+according to the configuration settings in .coafile
+
+## Proposed Change
+
+Here is the detailed implementation stepwise:
+
+1. We start by identifying different types of special commits namely revert,
+   merge and commits that disable ci builds.
+2. Each type of commit is then handled as a separate case.
+3. For commits that disable ci builds:
+   - Commit contents are checked to find `[ci skip]`.
+   - For such commits, it is checked that the commit body explains a valid
+     reason why this change should not include ci builds.
+   - Files modified in the commit should include files not covered by `.coafile`
+     or tests. If this is not the case, an error is raised to exclude `[ci skip]`
+     from the commit message.
+4. For merge commits:
+   - Merge commits are identified by checking the parents of the commit.
+   - In case the bear finds that the commit tries to merge master onto the pull
+     request, message is displayed to perform a rebase instead.
+5. For revert commits:
+   - Revert commits are identified based on the format of the commit.
+   - For revert commits, the patches of both commits are compared to ensure that
+     they are in fact reverse of each other. Minor differences are allowed.
+   - In case of total mismatch, a note in the commit message highlighting that
+     its not a clean revert is expected.
+   - A message requesting improvement in the commit message is displayed
```

Comment:
The bear will provide a 'fix' for the commit message?  I believe this is the 'ultimate' version, and you have issues about this.

Or an error message with suggestions in the message?  This is the easy solution.

Reply:
An error message with suggestions in the message. Like is does now for shortlog, body etc.